---
title: "R Notebook"
output: html_notebook
---

```{r}
library(Seurat)
library(dplyr)
library(Matrix)
library(gdata)
```

```{r}
setwd("C:/Users/aadit/OneDrive - IIT Delhi/Single Cell Sequencing/New folder (3)")
getwd()
```

```{r}
counts <- Read10X(data.dir = getwd())
```

```{r}
counts
```

```{r}
dim(counts)
```

```{r}
object.size(counts) # size in bytes
object.size(as.matrix(counts)) # size in bytes
```

```{r}
counts_per_cell <- Matrix::colSums(counts)
counts_per_gene <- Matrix::rowSums(counts)
genes_per_cell <- Matrix::colSums(counts>0)
cells_per_gene <- Matrix::rowSums(counts>0) 
```

```{r}
plot(sort(genes_per_cell), xlab='cell', log='y', main='genes per cell (ordered)')
```

```{r}
seurat<-CreateSeuratObject(counts, min.cells = 3, min.genes = 350, project = "scrna")
```

```{r}
mito.genes <- grep(pattern = "^MT-", x = rownames(x = GetAssayData(seurat)), value = TRUE)
```

```{r}
mito.genes
```

```{r}
percent.mito <- Matrix::colSums(GetAssayData(seurat, "counts")[mito.genes, ])/Matrix::colSums(GetAssayData(seurat, "counts"))
```

```{r}
percent.mito
```

```{r}
(seurat@meta.data)
```

```{r}
seurat <- AddMetaData(object = seurat, metadata = percent.mito, col.name = "percent.mito")
seurat@meta.data
```

```{r}
VlnPlot(object = seurat, features = c("nCount_RNA", "nFeature_RNA",  "percent.mito"), cols = 3, pt.size = 0.1)
```

```{r}
FeatureScatter(seurat, "nCount_RNA", "nFeature_RNA")
```

```{r}
str(seurat)
```

```{r}
VlnPlot(seurat, features = c("nFeature_RNA"), group.by = c('orig.ident'))
VlnPlot(seurat, features = c("percent.mito"), group.by = c('orig.ident'))
```

```{r}
seurat <- subset(seurat, subset = nFeature_RNA > 350 & nFeature_RNA < 5000 & percent.mito > -Inf & percent.mito < 0.1)
```

```{r}
seurat
```

```{r}
seurat <- NormalizeData(object = seurat, normalization.method = "LogNormalize", scale.factor = 1e4)
```

```{r}
seurat <- FindVariableFeatures(object = seurat, mean.function = ExpMean, dispersion.function = LogVMR, x.low.cutoff = 0.0125, x.high.cutoff = 1, y.cutoff = 0.5, num.bin=20)
```

```{r}
length(VariableFeatures(seurat))
```

```{r}
str(seurat)
```

```{r}
cc.genes <- readLines(paste0(getwd(),"/regev_lab_cell_cycle_genes.txt"))
```

```{r}
s.genes <- cc.genes[1:43]
g2m.genes <- cc.genes[44:97]
```

```{r}
seurat <- CellCycleScoring(seurat, s.features = s.genes, g2m.features = g2m.genes, set.ident = T)
```

```{r}
genes.dissoc <- c("ATF3", "BTG2", "CEBPB", "CEBPD", "CXCL3", "CXCL2", "CXCL1", "DNAJA1", "DNAJB1", "DUSP1", "EGR1", "FOS", "FOSB", "HSP90AA1", "HSP90AB1", "HSPA1A", "HSPA1B", "HSPA1A", "HSPA1B", "HSPA8", "HSPB1", "HSPE1", "HSPH1", "ID3", "IER2", "JUN", "JUNB", "JUND", "MT1X", "NFKBIA", "NR4A1", "PPP1R15A", "SOCS3", "ZFP36")
```

```{r}
seurat <- AddModuleScore(seurat, features = list(genes.dissoc), ctrl.size = 20, enrich.name = "genes_dissoc")
```

```{r}
str(seurat)
```

```{r}
FeatureScatter(seurat, "S.Score", "nFeature_RNA")
```
